### PR TITLE
tests: added tests for chi and debt edge cases in pile 

### PR DESCRIPTION
**Pull request by manuelpolzhofer**

### CODE REVIEW

File: src/test/pile.t.sol [hunk 1]
```diff
@@ -457,4 +457,59 @@ contract PileTest is DSTest {
...
+        pile.file(fee, fee);
+        hevm.warp(now + 1050 days); // 1,05 ^1050 = 1.7732257e+22
+
+        // init chi 10^27 *  1.7732257 * 10^22  ~ chi 10^49
+        // rdiv operation needs to mul chi with ONE (10^27)
+        // therefore: 10^49 * 10^27 = 10^76 < 1.1579209e+77
+        pile.drip(fee);
+    }
+
+    function testFailChiToHigh() public {
```
Comment (lucasvo) [inline src/test/pile.t.sol]:
```suggestion
    function testFailChiTooHigh() public {
```

File: src/test/pile.t.sol [hunk 2]
```diff
@@ -457,4 +457,59 @@ contract PileTest is DSTest {
...
+        // chi initial 10^27
+        uint fee = uint(1000000564701133626865910626); // 5 % / daily
+        pile.file(fee, fee);
+        hevm.warp(now + 1100 days); // 1,05 ^1100 = 2.0334288e+23
+
+        // init chi 10^27 *  2.0334288 * 10^23  ~ chi 10^50
+        // rdiv operation needs to mul chi with ONE (10^27)
+        // therefore: 10^50 * 10^27 = 10^77 same power as max value 1.1579209e+77
+        pile.drip(fee);
+
```
Comment (lucasvo) [inline src/test/pile.t.sol]:
```suggestion
```

File: src/test/pile.t.sol [hunk 3]
```diff
@@ -457,4 +457,59 @@ contract PileTest is DSTest {
...
+        uint principal = 1000000000  ether; // one billion 10^9 * 10^18 = 10^28
+        pile.file(loan, fee, 0);
+        borrow(loan, principal);
+        hevm.warp(now + 1050 days); // produces max ~ chi 10^49
+
+        // debt ~ 10^27 * 10^49 =  10^76 (max uint is 10^77)
+        pile.collect(loan);
+
+    }
+    function testFailDebtToHigh() public {
```
Comment (lucasvo) [inline src/test/pile.t.sol]:
```suggestion
    function testFailDebtTooHigh() public {
```
